src/common/updating/updating.cpp

```diff
@@ -212,32 +229,54 @@ namespace updating
-            download_dst /= new_version->msi_filename;
-            auto msi_installer_file_stream = co_await storage::Streams::FileRandomAccessStream::OpenAsync(download_dst.c_str(), storage::FileAccessMode::ReadWrite, storage::StorageOpenOptions::AllowReadersAndWriters, storage::Streams::FileOpenDisposition::CreateAlways);
-            co_await response.Content().WriteToStreamAsync(msi_installer_file_stream);
-            notifications::toast_params toast_params{ L"PTUpdateReadyTag", false };
+            std::filesystem::create_directories(installer_download_dst, _);
+            installer_download_dst /= new_version->msi_filename;
+
+
+            bool download_success = false;
+            for (size_t i = 0; i < MAX_DOWNLOAD_ATTEMPTS; ++i)
+            {
+                try
+                {
+                    co_await attempt_to_download_installer(installer_download_dst, new_version->msi_download_url);
+                    download_success = true;
+                    break;
+                }
+                catch (...)
+                {
+                    // reattempt to download or do nothing
+                }
+            }
+            if (!download_success)
+            {
+                notifications::toast_params toast_params{ UPDATE_NOTIFY_TOAST_TAG, false };
+                std::wstring contents = GITHUB_NEW_VERSION_DOWNLOAD_INSTALL_ERROR;
+                contents += current_version_to_next_version;
+
+                notifications::show_toast_with_activations(std::move(contents), {}, { notifications::link_button{ GITHUB_NEW_VERSION_AGREE, new_version->release_page_uri.ToString().c_str() } }, std::move(toast_params));
+
+                co_return;
+            }
+
+            notifications::toast_params toast_params{ UPDATE_READY_TOAST_TAG, false };
             std::wstring new_version_ready{ GITHUB_NEW_VERSION_READY_TO_INSTALL };
-            new_version_ready += L" ";
-            new_version_ready += new_version->version_string;
-            notifications::show_toast_with_activations(std::move(new_version_ready), {}, { notifications::link_button{ GITHUB_NEW_VERSION_UPDATE_NOW, L"powertoys://update_now/" }, notifications::link_button{ GITHUB_NEW_VERSION_UPDATE_AFTER_RESTART, L"powertoys://schedule_update/" }, notifications::snooze_button{ { { GITHUB_NEW_VERSION_UPDATE_SNOOZE_1D, 24 * 60 }, { GITHUB_NEW_VERSION_UPDATE_SNOOZE_5D, 120 * 60 } } } }, std::move(toast_params));
+            new_version_ready += current_version_to_next_version;
+
+            notifications::show_toast_with_activations(std::move(new_version_ready), {}, { notifications::link_button{ GITHUB_NEW_VERSION_UPDATE_NOW, L"powertoys://update_now/" }, notifications::link_button{ GITHUB_NEW_VERSION_UPDATE_AFTER_RESTART, L"powertoys://schedule_update/" }, notifications::snooze_button{ GITHUB_NEW_VERSION_SNOOZE_TITLE, { { GITHUB_NEW_VERSION_UPDATE_SNOOZE_1D, 24 * 60 }, { GITHUB_NEW_VERSION_UPDATE_SNOOZE_5D, 120 * 60 } } } }, std::move(toast_params));
```

Comment:
This line have 449 characters. It should be indented eg.

            notifications::show_toast_with_activations(std::move(new_version_ready),
                                                       {},
                                                       { notifications::link_button{ GITHUB_NEW_VERSION_UPDATE_NOW, L"powertoys://update_now/" },
                                                         notifications::link_button{ GITHUB_NEW_VERSION_UPDATE_AFTER_RESTART, L"powertoys://schedule_update/" },
                                                         notifications::snooze_button{ GITHUB_NEW_VERSION_SNOOZE_TITLE, { { GITHUB_NEW_VERSION_UPDATE_SNOOZE_1D, 24 * 60 }, { GITHUB_NEW_VERSION_UPDATE_SNOOZE_5D, 120 * 60 } } } },
                                                       std::move(toast_params));

Reply:
I agree, but we currently use the .clang-format file which doesn't set the max line length, so it will be reverted on the next format. I'd suggest we set a line length limit. @enricogior thoughts?

Reply:
If you break the line after comma(,) Clang will respect that.
Try pasting my proposition and formatting